modules/inject/src/main/scala/io/janstenpickle/trace4cats/inject/package.scala

```diff
@@ -0,0 +1,11 @@
+package io.janstenpickle.trace4cats
+
+import cats.data.Kleisli
+import cats.effect.Resource
+import io.janstenpickle.trace4cats.model.{SpanKind, TraceHeaders}
+
+package object inject {
+  type SpanName = String
+  type SpanParams = (SpanName, SpanKind, TraceHeaders)
+  type ResourceReader[F[_], -In, Out] = Kleisli[Resource[F, *], In, Out]
```

Comment:
This is fine. Firstly I wanted to suggest the same, but I overlooked the `tapWithF` operator and also I thought you wanted to have a new type to add some combinators missing in Kleisli.

I also think it's safe to write 
```scala
type ResourceReader[F[_], -In, +Out] = Kleisli[Resource[F, +*], In, Out @uncheckedVariance]]
```
since `Resource` is covariant.

It's not nitpicking, but I also like `ResourceKleisli` name much better than `ResourceReader`, because `ReaderT` and `Kleisli` are semantically different, despite isomorphism between their representations in `cats`. We consider _this thing_ an arrow, a function from `In` to `Resource[F, Out]` (thus `Kleisli`), not just a monadic value `Resource[F, Out]` with a static monadic context `In`.

Reply:
Thanks for the top on the covariance.

I was 50/50 on whether to call it `Kleisli` too, but your reasoning about the semantic differences is a good reason to change.